### PR TITLE
fix(argo-workflows): avoid label exceeding maximum length

### PR DESCRIPTION
**Pull request by steenzout**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [X] Any new values are backwards compatible and/or have sensible default.
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/argo-workflows/templates/_helpers.tpl [hunk 1]
```diff
@@ -45,6 +45,32 @@ Create chart name and version as used by the chart label.
...
 {{- end -}}
 
+{{/*
+Create kubernetes friendly chart version label for the controller.
+Examples:
+image.tag = v3.4.4
+output    = v3.4.4
+
+image.tag = v3.4.4@sha256:d06860f1394a94ac3ff8401126ef32ba28915aa6c3c982c7e607ea0b4dadb696
+output    = v3.4.4
```
Comment (yu-croco) [inline charts/argo-workflows/templates/_helpers.tpl:55]: nice examples 👍

File: charts/argo-workflows/templates/_helpers.tpl [hunk 2]
```diff
@@ -45,6 +45,32 @@ Create chart name and version as used by the chart label.
...
+*/}}
+{{- define "argo-workflows.server_chart_version_label" -}}
+{{- regexReplaceAll "[^a-zA-Z0-9-_.]+" (regexReplaceAll "@sha256:[a-f0-9]+" (default (include "argo-workflows.defaultTag" .) .Values.server.image.tag) "") "" | trunc 63 | quote -}}
+{{- end -}}
+
 {{/*
 Common labels
 */}}
```
Comment (steenzout) [inline charts/argo-workflows/templates/_helpers.tpl:76]: curious: is there a reason why this chart doesn't set the `app.kubernetes.io/version` in common labels like, for example, the `argo-workflows`chart?

Comment (steenzout) [reply charts/argo-workflows/templates/_helpers.tpl:76]: figured it out.

**Review by jmeridth — APPROVED** [review]

**Review by yu-croco — APPROVED** [review]
Thank you for your contribution! LGTM

**Pull request merged**